src/modules/useProtectedRoute.tsx

```diff
@@ -0,0 +1,32 @@
+import { useUser } from "@/modules/common/hooks/useUserStore";
+import { roleToRoleNumber, type Role } from "@/modules/services/users";
+import { useRouter } from "next/router";
+import { useEffect } from "react";
+
+type UseProtectedRouteArgs = {
+  minimumRole?: Role;
+  redirectTo?: string;
+};
+
+const useProtectedRoute = (args?: UseProtectedRouteArgs) => {
+  const { minimumRole, redirectTo = "/login" } = args ?? {};
+
+  const router = useRouter();
+  const { data: user, isSuccess, isError } = useUser();
+
+  useEffect(() => {
+    if (!minimumRole) {
+      return;
+    }
+
+    if ((isSuccess || isError) && !user) {
```

Comment:
Does we need `(isSuccess || isError)`

Reply:
must use because to make sure that api is already get user and also have no user

Reply:
> Does we need `(isSuccess || isError)`

must use because to make sure that api is already get user and also have no user

Reply:
The Params and Args types define the structure of objects that might be passed as arguments to a function. You can use either Params or Args, as the choice of name is up to your preference. 

from chatgpt